runtime/smithy-test/common/src/aws/smithy/kotlin/runtime/smithy/test/HttpRequestTest.kt

```diff
@@ -64,7 +67,15 @@ public fun httpRequestTest(block: HttpRequestTestBuilder.() -> Unit): TestResult
             testHeaders["Content-Length"] = contentLength.toString()
         }
 
-        actual = HttpRequest(method = request.method, url = request.url, headers = testHeaders.build(), body = request.body)
+        val body = if (testBuilder.expected.bodyMediaType?.isBinaryMediaType == true) {
+            runBlocking { request.body.readAll() }?.let {
+                HttpBody.fromBytes(it.encodeBase64())
+            } ?: request.body
+        } else {
+            request.body
+        }
```

Comment:
**Question**: Why is `runBlocking` necessary here? We're inside of `runTest` which has a coroutine scope.

Reply:
It was necessary because the `TestEngine` `roundTripImpl` parameter was not a suspend function, I updated that and removed the `runBlocking`